src/Gyro.cpp

```diff
@@ -34,67 +42,88 @@ void Gyro::startup()
         return;
     }
 
-    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1
+    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1 to disable sleep and set 8MHz clock
     Wire.beginTransmission(MPU6050_I2C_ADDR);
     Wire.write(MPU6050_REG_PWR_MGMT_1);
-    Wire.write(0);      // Disable sleep, 8 MHz clock
+    Wire.write(0); // Disable sleep, 8 MHz clock
     Wire.endTransmission(true);
 
-    // Execute 1 byte write to MPU6050_REG_PWR_MGMT_1
+    // Execute 1 byte read/write to MPU6050_REG_ACCEL_CONFIG to set 4g sensititvity
+    Wire.beginTransmission(MPU6050_I2C_ADDR);
+    Wire.write(MPU6050_REG_ACCEL_CONFIG);
+    Wire.endTransmission();
+    Wire.requestFrom(MPU6050_I2C_ADDR, 1);
+    byte x = Wire.read(); //the value of Register-28 is in x
+    x = (x & 0b11100111);     //clear values of Bit4 and Bit3 fo 2g sensitivity
+    Wire.beginTransmission(MPU6050_I2C_ADDR);
+    Wire.write(MPU6050_REG_ACCEL_CONFIG);
+    Wire.write(x);
+    Wire.endTransmission();
+
+    // Execute 1 byte write to MPU6050_REG_CONFIG to set slowest refresh rate
     Wire.beginTransmission(MPU6050_I2C_ADDR);
     Wire.write(MPU6050_REG_CONFIG);
-    Wire.write(6);      // 5Hz bandwidth (lowest) for smoothing
+    Wire.write(6); // 5Hz bandwidth (lowest) for smoothing
     Wire.endTransmission(true);
 
     LOGV1(DEBUG_INFO, F("GYRO:: Started"));
 }
 
+// Shuts down the MPU-6050 device.
+// Currently does nothing.
 void Gyro::shutdown()
-/* Shuts down the MPU-6050 device.
-   Currently does nothing.
-*/
 {
     LOGV1(DEBUG_INFO, F("GYRO: Shutdown"));
     // Nothing to do
 }
 
+// Returns roll & tilt angles from MPU-6050 device in angle_t object in degrees.
+// If MPU-6050 is not found then returns {0,0}.
 angle_t Gyro::getCurrentAngles()
-/* Returns roll & tilt angles from MPU-6050 device in angle_t object in degrees.
-   If MPU-6050 is not found then returns {0,0}.
-*/
 {
-    const int windowSize = 16;
+    const int windowSize = 8;
     // Read the accelerometer data
     struct angle_t result;
     result.pitchAngle = 0;
     result.rollAngle = 0;
     if (!isPresent)
-        return result;     // Gyro is not available
+        return result; // Gyro is not available
 
     for (int i = 0; i < windowSize; i++)
     {
         // Execute 6 byte read from MPU6050_REG_WHO_AM_I
         Wire.beginTransmission(MPU6050_I2C_ADDR);
-        Wire.write(MPU6050_REG_ACCEL_XOUT_H); 
+        Wire.write(MPU6050_REG_ACCEL_XOUT_H);
         Wire.endTransmission(false);
-        Wire.requestFrom(MPU6050_I2C_ADDR, 6, 1);  // Read 6 registers total, each axis value is stored in 2 registers
-        int16_t AcX = Wire.read() << 8 | Wire.read(); // X-axis value
-        int16_t AcY = Wire.read() << 8 | Wire.read(); // Y-axis value
-        int16_t AcZ = Wire.read() << 8 | Wire.read(); // Z-axis value
+        Wire.requestFrom(MPU6050_I2C_ADDR, 6, 1);     // Read 6 registers total, each axis value is stored in 2 registers
+        uint8_t b0 =  Wire.read();
```

Comment:
I'd rename to `AxXLow`/`AcXHigh` or something to be more explicit, but up to you. (I would also pepper `const` everywhere but that's just me :grin: )

Reply:
Fixed